crates/pallet-erc20-support/src/lib.rs

```diff
@@ -89,6 +105,33 @@ pub mod pallet {
         /// Spender can't transfer tokens more than allowed.
         SpendMoreThanAllowed,
     }
+
+    #[pallet::hooks]
+    impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
+        fn on_runtime_upgrade() -> Weight {
+            let mut weight = T::DbWeight::get().reads(1);
+
+            if StorageVersion::get::<Pallet<T, I>>() == 0 {
+                weight.saturating_accrue(migrations::v1::migrate::<T, I>());
+                StorageVersion::new(1).put::<Pallet<T, I>>();
+                weight.saturating_accrue(T::DbWeight::get().writes(1));
+            }
+
+            assert_eq!(<Pallet<T, I>>::on_chain_storage_version(), STORAGE_VERSION);
```

Comment:
I meant for this to be at the `post_upgrade` check

Reply:
Also don't use `STORAGE_VERSION` directly, use `<Pallet<T, I>>::current_storage_version()` instead

Reply:
https://github.com/humanode-network/humanode/pull/809/commits/6a5072127188411bfea8fce855652917d1553d25